File: src/main/java/com/45secondapps/service/AppService.java
package com.45secondapps.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.45secondapps.domain.App;
import com.45secondapps.dto.AppDTO;
import com.45secondapps.dto.AppSearchDTO;
import com.45secondapps.dto.AppPageDTO;
import com.45secondapps.dto.AppConvertCriteriaDTO;
import com.45secondapps.service.GenericService;
import com.45secondapps.dto.common.RequestDTO;
import com.45secondapps.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AppService extends GenericService<App, Integer> {

	List<App> findAll();

	ResultDTO addApp(AppDTO appDTO, RequestDTO requestDTO);

	ResultDTO updateApp(AppDTO appDTO, RequestDTO requestDTO);

    Page<App> getAllApps(Pageable pageable);

    Page<App> getAllApps(Specification<App> spec, Pageable pageable);

	ResponseEntity<AppPageDTO> getApps(AppSearchDTO appSearchDTO);
	
	List<AppDTO> convertAppsToAppDTOs(List<App> apps, AppConvertCriteriaDTO convertCriteria);

	AppDTO getAppDTOById(Integer appId);







}

File: 45secondapps_ui/src/services/AppService.js
import http from "../http-common"; 

class AppService {
  getAllApps(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/app/apps`, searchDTO);
  }

  get(appId) {
    return this.getRequest(`/app/${appId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/app?field=${matchData}`, null);
  }

  addApp(data) {
    return http.post("/app/addApp", data);
  }

  update(data) {
  	return http.post("/app/updateApp", data);
  }
  
  uploadImage(data,appId) {
  	return http.postForm("/app/uploadImage/"+appId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AppService();
